src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static bool IsLiteralNull(this BoundExpression node)
 
         public static bool IsLiteralDefault(this BoundExpression node)
         {
-            return node.Kind == BoundKind.DefaultLiteral && (object)node.Type == null;
+            return node.Kind == BoundKind.DefaultLiteral && node.Syntax.Kind() == SyntaxKind.DefaultLiteralExpression;
         }
```

Comment:
`BoundKind.DefaultLiteral` used to be `BoundKind.DefaultOperator`. Maybe I will rename it back.
But to answer your question, `IsLiteralDefault` should return `false` for `default(int)`. It should only return `true` for the `default` literal.